pkg/downloader/chart_downloader.go

```diff
@@ -97,7 +97,16 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 
 	name := filepath.Base(u.Path)
 	destfile := filepath.Join(dest, name)
-	if err := ioutil.WriteFile(destfile, data.Bytes(), 0644); err != nil {
+	tmpfile, err := ioutil.TempFile(dest, "tmp")
+	if err != nil {
+		return "", nil, err
+	}
```

Comment:
If there is a failure between the creation of the tmpfile and the `os.Rename` command, I believe that the `tmpfile` will never be removed (because it is the user's responsibility to clean up files created by `ioutil.TempFile`). Not sure if this is an actual problem, but if so, I think we could solve with a 

```
defer func() {
    if _, err := os.State(tmpfile.Name()); err == nil {
        os.Remove(tmpfileName())
    }
}